app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedPresenter.java

```diff
@@ -0,0 +1,175 @@
+    private MainApplication mainApplication;
+    private NewsStorage newsStorage;
+
+    NewsFeedPresenter(NewsFeedContract.View tasksView, MainApplication mainApplication) {
+        view = checkNotNull(tasksView, "tasksView cannot be null!");
+        view.setPresenter(this);
+        this.mainApplication = mainApplication;
+    }
+
+    @Override
+    public void start() {
+        initNewsItemListCallback();
+        initNewsStorage();
+        Log.d(TAG, "start: " + showOnlyFavorite);
+        view.setFavoriteIcon(showOnlyFavorite);
+        view.setCategoryTitle(category);
+        loadNewsFromApi();
+    }
+
+    private void loadNewsFromApi() {
+        Log.d(TAG, "loadNewsFromApi: ");
+        view.showProgressBar();
+        performCall();
+    }
+
+    private void loadNewsFromDB() {
+        Log.d(TAG, "loadNewsFromDB: ");
+        news = getNewsFromDB();
+    }
+
+    @Override
+    public void openNewsDetails(int position) {
+        showNewsInBrowserByUrl(position);
+    }
+
+    @Override
+    public void changeNewsFavoriteState(int position) {
+        NewsItem item = news.get(position);
+        item.invertFavoriteState();
+        newsStorage.updateNews(item);
+        if (showOnlyFavorite && !item.isFavorite()){
+            news.remove(position);
+            view.removeNewsItem(position);
+            if (news.isEmpty()){
+                view.updateNewsListUI(null);
+            }
+        } else {
+            view.updateNewsItem(position);
+        }
+    }
+
+    @Override
+    public void setCategory(Category category) {
+        Log.d(TAG, "setCategory: ");
+        this.category = category;
+    }
+
+    @Override
+    public Category getCategory() {
+        return this.category;
+    }
+
+    @Override
+    public void setShowFavorite(Boolean showOnlyFavorite) {
+        Log.d(TAG, "setShowFavorite: " + showOnlyFavorite);
+        this.showOnlyFavorite = showOnlyFavorite;
+    }
+
+    @Override
+    public Boolean getShowFavorite() {
+        return this.showOnlyFavorite;
+    }
+
+    @Override
+    public void invertFavoriteState() {
+        showOnlyFavorite = !showOnlyFavorite;
+        Log.d(TAG, "invertFavoriteState: " + showOnlyFavorite);
+        view.setFavoriteIcon(showOnlyFavorite);
+        loadNewsFromDB();
+        view.updateNewsListUI(news);
+    }
+
+    @Override
+    public void updateNewsFromApi() {
+        Log.d(TAG, "updateNewsFromApi: ");
+        loadNewsFromApi();
+    }
+
+    private void showNewsInBrowserByUrl(int position) {
+        NewsItem item = news.get(position);
+        String url = item.getUrl();
+        view.showNewsDetailsUI(url);
+    }
+
+    private List<NewsItem> getNewsFromDB() {
+        return newsStorage.getNewsFromDB(showOnlyFavorite, category);
+    }
+
+    private void initNewsStorage() {
+        NewsItemDAO newsItemDAO = mainApplication.getNewsDataBase().newsItemDAO();
+        newsStorage = new NewsStorage(newsItemDAO);
```

Comment:
Can't we move this initialization to the storage constructor and just pass a context there?